mantidproject/mantid
# Implemented an iterator for `DetectorInfo`

**Description of work.**
Implemented an iterator for `DetectorInfo` for both C++ and Python. 
Added basic units tests for C++ and for Python.
 
**Report to:** @OwenArnold. 

**To test:**
Not much to test - please ensure that both sets of unit tests pass.


Run `ctest -C Debug -R DetectorInfoIteratorTest -V` to only run the C++ unit tests.
Run `ctest -C Debug -R DetectorInfoTest -V` to run the Python unit tests.

Fixes #23145 . 




---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
